File: copilot-demo-labs/use-cases/03-debugging-simple/src/config_loader.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace Demo;

public static class ConfigLoader
{
    public static Dictionary<string, string> LoadFlags(string path)
    {
        var flags = new Dictionary<string, string>();

        foreach (var line in File.ReadLines(path))
        {
            if (!line.Contains('='))
            {
                continue;
            }

            var parts = line.Split('=', 2);
            flags[parts[0].Trim()] = parts[1].Trim();
        }

        throw new InvalidOperationException("flag is not defined");
    }
}

File: copilot-demo-labs/use-cases/03-debugging-simple/src/config_loader.py
def load_flags(path: str) -> dict[str, str]:
    flags: dict[str, str] = {}

    with open(path, "r", encoding="utf-8") as f:
        for line in f:
            if "=" in line:
                k, v = line.split("=", 1)
                flags[k.strip()] = v.strip()

    return flag
